rmc/utils/constraint.py

```diff
@@ -1310,45 +1318,350 @@ def create_context_with_oe(
+        _read_if_exists=not overwrite_temp,
+        overwrite=overwrite_temp,
+    )
+    missing_ht = missing_ht.annotate(
+        start_aa=hl.if_else(
+            hl.is_missing(missing_ht.start_aa),
+            filt_context_ht[missing_ht.closest_start_mis, missing_ht.transcript].ref_aa,
+            missing_ht.start_aa,
+        ),
+        stop_aa=hl.if_else(
+            hl.is_missing(missing_ht.stop_aa),
+            filt_context_ht[missing_ht.closest_stop_mis, missing_ht.transcript].ref_aa,
+            missing_ht.stop_aa,
+        ),
+    )
+    return missing_ht.select("start_aa", "stop_aa")
+
+
+def check_and_fix_missing_aa(
+    ht: hl.Table, context_ht: hl.Table, overwrite_temp: bool
+) -> hl.Table:
+    """
+    Check for and fix any missing amino acid information.
+
+    :param ht: Input Table. Should be RMC results HT annotated with amino acid information.
+    :param context_ht: VEP context HT filtered to keep only transcript ID, protein number, and amino acid information.
+    :param overwrite_temp: Whether to overwrite temporary data.
+        If False, will read existing temp data rather than overwriting.
+        If True, will overwrite temp data.
+    :return: RMC results HT annotated with amino acids for all region start/stops.
+    """
+    missing_ht = ht.filter(hl.is_missing(ht.start_aa) | hl.is_missing(ht.stop_aa))
+    missing_start_or_stop = missing_ht.count()
+    if missing_start_or_stop == 0:
+        return ht
+
+    logger.info(
+        "%i sites are missing either their stop or start AA!", missing_start_or_stop
+    )
+
+    # Get strand from browser HT
+    # TODO: recreate browser HT to include strand annotation
+    browser_ht = gene_model.ht().select("strand")
+    ht = ht.annotate(strand=browser_ht[ht.transcript].strand)
+
+    # Get CDS start/stops from CDS HT
+    transcript_ht = transcript_cds.ht().key_by()
+    transcript_ht = transcript_ht.annotate(
+        cds_start=transcript_ht.interval.start,
+        cds_stop=transcript_ht.interval.end,
+    )
+    ht = ht.annotate(**transcript_ht[ht.transcript])
+    ht = ht.annotate(
+        transcript_start=ht.start_coordinate == ht.cds_start,
+        transcript_stop=ht.stop_coordinate == ht.cds_stop,
+    )
+    transcript_start_stop_fix_ht = fix_transcript_start_stops(ht, overwrite_temp)
+    ht = join_and_fix_aa(ht, transcript_start_stop_fix_ht)
+    ht = ht.checkpoint(
+        f"{TEMP_PATH_WITH_FAST_DEL}/rmc/RMC_results_transcript_start_stop_aa_fix.ht",
+        _read_if_exists=not overwrite_temp,
+        overwrite=overwrite_temp,
+    )
+    region_fix_ht = fix_region_start_stops(ht, context_ht, overwrite_temp)
+    ht = join_and_fix_aa(ht, region_fix_ht)
+    ht = ht.checkpoint(
+        f"{TEMP_PATH_WITH_FAST_DEL}/rmc/RMC_results_all_aa_fix.ht",
+        _read_if_exists=not overwrite_temp,
+        overwrite=overwrite_temp,
+    )
+    return ht
+
+
+def annotate_transcript_globals(ht: hl.Table) -> hl.Table:
```

Comment:
```suggestion
def add_globals_rmc_browser(ht: hl.Table) -> hl.Table:
```
Suggestion for function name